proxy/proxy.go

```diff
@@ -1132,6 +1124,8 @@ func (p *Proxy) do(ctx *context) (err error) {
 		backendStart := time.Now()
 		rsp, perr := p.makeBackendRequest(ctx, backendContext)
 		if perr != nil {
+			p.makeErrorResponse(ctx, perr)
```

Comment:
Lets move this close to the return in the else branch

Reply:
done